test/SlowTests/Server/Documents/TimeSeries/TimeSeriesTombstoneCleaner.cs

```diff
@@ -358,6 +359,94 @@ public async Task CleanTimeSeriesTombstonesInTheClusterWithOnlyFullBackup()
+                var users = new List<string>()
+                {
+                    "user/1",
+                    "user/2",
+                    "user/11",
+                    "user/21",
+                    "user/12",
+                    "user/22",
+                    "user/3",
+                    "user/13"
+                };
+
+                using (var session = store1.OpenSession())
+                {
+                    foreach (string user in users)
+                    {
+                        session.Store(new User {Name = "EGR"}, user);
+                    }
+
+                    session.SaveChanges();
+                }
+
+                foreach (string user in users)
+                {
+                    using (var session = store1.OpenSession())
+                    {
+                        var tsf = session.TimeSeriesFor(user, "raven");
+                        for (int i = 0; i <= 30; i++)
+                        {
+                            tsf.Append(_baseline.AddDays(i), i);
+                        }
+
+                        session.SaveChanges();
+                    }
+                }
+
+                foreach (string user in users)
+                {
+                    using (var session = store1.OpenSession())
+                    {
+                        var tsf = session.TimeSeriesFor(user, "raven");
+                        tsf.Delete(_baseline, _baseline.AddDays(29));
+                        session.SaveChanges();
+                    }
+                }
+
+                var storage = await Server.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(store1.Database);
+                var cleaner = storage.TombstoneCleaner;
+
+                await SetupReplicationAsync(store1, store2);
+                await EnsureReplicatingAsync(store1, store2);
+
+                using (var controller = new ReplicationController(storage))
+                {
+                    foreach (string user in users)
+                    {
+                        using (var session = store1.OpenSession())
+                        {
+                            var tsf = session.TimeSeriesFor(user, "raven");
+                            tsf.Delete(_baseline.AddDays(29), _baseline.AddDays(29));
+                            tsf.Delete(_baseline.AddDays(30), _baseline.AddDays(30));
+                            session.SaveChanges();
+                        }
+                    }
+
+                    controller.ReplicateOnce();
+
+                    await cleaner.ExecuteCleanup();
+                }
+
+                await EnsureReplicatingAsync(store1, store2);
+
+                var tss = storage.DocumentsStorage.TimeSeriesStorage;
+
+                using (storage.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext ctx))
+                using (ctx.OpenReadTransaction())
+                {
+                    Assert.True(tss.Stats.GetNumberOfEntries(ctx) >= 6);
```

Comment:
why >=6? this should be deterministic no?
also can we run the cleaner and assert for 0 ?

Reply:
done